### DISCUSSION TITLE
How should errors thrown during extension activation be handled?

### DISCUSSION THREAD

**Discussion by t1m0thyj** [microsoft/vscode-discussions#231] [Extension Development QnA] [answered]
When a VS Code extension fails to activate because its `activate` method threw an error, this results in an undesirable state:
1. The activation error is written to the Extension Host log, but apart from this there is no error message shown to explain the failure, so most users will have no idea that the extension activation failed.
2. The extension is left in a half-broken state, where anything it contributes like commands and sidebar views are still present in the VS Code window. When the user activates one of these commands from the command palette or by clicking a button in the empty sidebar view, these actions will fail.

Instead of these behaviors, I would expect VS Code to do the following:
1. Show a warning message in the Notifications tray informing the user that activating an extension failed.
2. Unsubscribe all the extension's contributions (and perhaps disable it?) so that the UI is not left in this half-broken state.

Before opening an issue on the VS Code repo about this, I want to know whether others think these expectations are valid or if I misunderstand how extensions should behave. Is it my responsibility as a VS Code extension developer to catch any errors that may happen during my extension's `activate` method, and implement the above behaviors (showing a warning message and unsubscribing contributions) myself?

**Comment by isidorn** [comment]
I know @lramos15 is looking into improving the extension error capturing API so he might have more context here.

**Reply by lramos15** [reply, marked as answer]
Telemetry API will not help if the extension fails to activate as the code will not be executing.

**Reply by burkeholland** [reply]
Marking as answer as this is several months old. If you feel this is a mistake, please feel free to re-open.

**Reply by lramos15** [reply]
You may be also interested in https://github.com/microsoft/vscode/issues/179633 which was recently created to track some of the pain points you mentioned
